casa/IO/test/tByteSinkSource.cc

```diff
@@ -171,16 +169,16 @@ int main ()
 	cout << testString    << endl;
     }
     {
-	RegularFileIO regularFileIO (Path("tByteSinkSource_tmp.dat1"),
-				     ByteIO::Update);
-	CanonicalIO canonicalIO(&regularFileIO, 5);
-	ByteSinkSource sinkSource(&canonicalIO);
+        std::shared_ptr<ByteIO> regularFileIO
+          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::Update));
+        std::shared_ptr<TypeIO> canonicalIO(new CanonicalIO(regularFileIO, 5));
```

Comment:
```suggestion
        std::shared_ptr<RegularFileIO> regularFileIO
          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::Update));
        std::shared_ptr<CanonicalIO> canonicalIO(new CanonicalIO(regularFileIO, 5));
```
Additionally, you could use `std::make_shared` here.